monitor.py

```diff
@@ -843,21 +847,30 @@ def parse_entry(self, line: bytes) -> MutableMapping[str, Any]:  # noqa: C901, C
                         # So it's *from* the ship
                         self.state['Cargo'][name] -= c['Count']
 
-            elif event_type == 'ShipLockerMaterials':
+            elif event_type == 'ShipLocker':
+                # As of 4.0.0.400 (2021-06-10)
+                # "ShipLocker" will be a full list written to the journal at startup/boarding/disembarking, and also
+                # written to a separate shiplocker.json file - other updates will just update that file and mention it
+                # has changed with an empty shiplocker event in the main journal.
+
+                if not all([entry.get(t, False) for t in ('Components', 'Consumables', 'Data', 'Items')]):
```

Comment:
As ShipLocker always writes to the file _anyway_, possibly just always read it?

Additionally, all() can take a generator
```suggestion
                if not all(entry.get(t, False) for t in ('Components', 'Consumables', 'Data', 'Items')):
```

Reply:
Eh, if the data is in the event I'm happy just using that, save the file open/read.  I'll remove those `[]` then.

Reply:
OK, I relented, we nose always load the file.